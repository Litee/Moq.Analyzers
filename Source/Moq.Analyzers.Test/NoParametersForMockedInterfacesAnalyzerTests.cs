using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.CodeAnalysis.Diagnostics;
using System.IO;
using TestHelper;
using Xunit;

namespace Moq.Analyzers.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class NoParametersForMockedInterfacesAnalyzerTests : DiagnosticVerifier
    {

        [Fact]
        public void ShouldFailIfMockedInterfaceHasParameters()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/MockInterfaceWithParameters.cs")));
        }

        [Fact]
        public void ShouldPassIfCustomMockClassIsUsed()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/MockInterfaceWithParametersCustomMockFile.cs")));
        }

        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
        {
            return new NoParametersForMockedInterfacesAnalyzer();
        }
    }
}