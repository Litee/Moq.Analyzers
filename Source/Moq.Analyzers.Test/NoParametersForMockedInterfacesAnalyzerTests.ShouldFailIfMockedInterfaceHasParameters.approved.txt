Diagnostic 1
	Id: Moq1001
	Location: SourceFile(Test0.cs[241..251))
	Code: (25, true)
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 2
	Id: Moq1001
	Location: SourceFile(Test0.cs[298..305))
	Code: ("123")
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 3
	Id: Moq1001
	Location: SourceFile(Test0.cs[380..390))
	Code: (25, true)
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 4
	Id: Moq1001
	Location: SourceFile(Test0.cs[469..476))
	Code: ("123")
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 5
	Id: Moq1001
	Location: SourceFile(Test0.cs[576..609))
	Code: (Moq.MockBehavior.Default, "123")
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 6
	Id: Moq1001
	Location: SourceFile(Test0.cs[656..687))
	Code: (MockBehavior.Strict, 25, true)
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 7
	Id: Moq1001
	Location: SourceFile(Test0.cs[762..795))
	Code: (Moq.MockBehavior.Default, "123")
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 8
	Id: Moq1001
	Location: SourceFile(Test0.cs[874..904))
	Code: (MockBehavior.Loose, 25, true)
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

